util/reflect.go

```diff
@@ -95,6 +101,12 @@ func PtrToValue(value reflect.Value) reflect.Value {
 	return value
 }
 
+// StructValueHasNFields returns true if the reflect.Value representing a struct
+// v has n fields.
+func StructValueHasNFields(v reflect.Value, n int) bool {
```

Comment:
The functions in util.reflect are meant to be panic free - suggest renaming this to IsValueStructWithNFields if you want to keep the bool return value and add check for struct.

Reply:
Done.